src/Tools/Tra/paw_tra.f90

```diff
@@ -2349,6 +2351,46 @@ SUBROUTINE WRITEXYZ(NFIL,FRAME,NAT,ID,R)
       RETURN
       END
 !
+!     ...1.........2.........3.........4.........5.........6.........7.........8
+      SUBROUTINE WRITEEXTXYZ(NFIL,FRAME,NAT,ID,R,CELL,TIME)
+!     ******************************************************************
+!     **                                                              **
+!     ******************************************************************
+      USE STRINGS_MODULE
+      INTEGER(4)  ,INTENT(IN) :: NFIL
+      INTEGER(4)  ,INTENT(IN) :: FRAME
+      INTEGER(4)  ,INTENT(IN) :: NAT
+      CHARACTER(2),INTENT(IN) :: ID(NAT)
+      REAL(8)     ,INTENT(IN) :: R(3,NAT)
+      REAL(8)     ,INTENT(IN) :: CELL(9)
+      REAL(8)     ,INTENT(IN) :: TIME
+      INTEGER(4)              :: IAT
+      REAL(8)                 :: ANGSTROM, PICO, SECOND           
+      CHARACTER(100)          :: STRING
+      CHARACTER(200)          :: EXTXYZ
+      CHARACTER(2)            :: SPECIES
+!     ******************************************************************
+      CALL CONSTANTS$GET('ANGSTROM',ANGSTROM)
+      CALL CONSTANTS$GET('PICO', PICO)
+      CALL CONSTANTS$GET('SECOND', SECOND)
+      IF(FRAME.EQ.1) REWIND NFIL
+      WRITE(NFIL,*)NAT
+      WRITE(STRING,FMT='(9F10.5)')CELL/ANGSTROM
+      EXTXYZ='Lattice="'//TRIM(ADJUSTL(STRING))//'" Properties=species:S:1:pos:R:3 Iter='
+      WRITE(STRING,FMT='(I10)')FRAME
+      EXTXYZ=TRIM(EXTXYZ)//TRIM(ADJUSTL(STRING))//' Time='
+      WRITE(STRING,FMT='(F10.5)')TIME/(PICO*SECOND)
+      EXTXYZ=TRIM(EXTXYZ)//TRIM(ADJUSTL(STRING))
+      WRITE(NFIL,*)EXTXYZ
+      DO IAT=1,NAT
+        ! TRANSFORM ELEMENT SYMBOL TO UPPERCASE+LOWERCASE
+        SPECIES=+ID(IAT)
+        SPECIES(2:2)=-SPECIES(2:2)
```

Comment:
Here the species type (Element symbol) is extracted from the atom name. It is an optional feature to incorporate the element symbol in the atom name, but we cannot rely on it. Either the connection must be mandatory throughout the code, or we need to directly access the element symbol/atom type. This is important to be able to define artificial elements (empty atom or fractional atomic number.)
This needs a general review of the code and a strategy decision (->Peter)